pkg/checkers/checker.go

```diff
@@ -0,0 +1,98 @@
+package checkers
+
+import (
+	"github.com/google/shlex"
+	"github.com/kubereboot/kured/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+)
+
+// Checker is the standard interface to use to check
+// if a reboot is required. Its types must implement a
+// CheckRebootRequired method which returns a single boolean
+// clarifying whether a reboot is expected or not.
+type Checker interface {
+	CheckRebootRequired() bool
```

Comment:
`CheckRebootRequired()` as a method for `Checker` has a bit of a stutter, maybe just `RebootRequired()` is sufficient